tests/python/unittest/test_gluon.py

```diff
@@ -966,6 +968,12 @@ def test_save_load():
 
     net.load_params('test.params')
 
+@unittest.skip("fails due to mkldnn issue. renable after mkldnn fixed")
+def test_hybrid_multi_context():
+    net = mx.gluon.model_zoo.vision.get_resnet(1, 18)
+    net.initialize(ctx=[mx.cpu(0), mx.cpu(1)])
+    net.hybridize()
+    net(mx.nd.zeros((1, 3, 32, 32), ctx=mx.cpu(0))).asnumpy()
```

Comment:
does this mean it would be broken for mkldnn?
cc'd @zheng-da

Reply:
I created a PR to fix this issue.

Reply:
@zheng-da @piiswrong let's have all the changes here instead.